index.js

```diff
@@ -55,12 +68,18 @@ module.exports = function (mode) {
       }
       files = statusFiles;
 
-      if (mode === 'D') {
-        // Deleted files. Make into vinyl files
-        files.map(makeVinylFile).forEach(function (file) {
-          stream.push(file);
-        });
-      }
+      // Deleted files. Make into vinyl files
+      _.chain(files)
+        .map(function(file) {
+          if (file.mode !== 'D') return;
+          return file.path;
+        })
+        .remove()
```

Comment:
also in this case `.map().remove()` could be replaced by `.filter().pluck()` to clarify your intentions  
```javascript
.filter(function(file) {
  return file.mode === 'D';
})
.pluck('path')
```